x-pack/plugins/aiops/public/hooks/use_aiops_app_context.ts

```diff
@@ -22,6 +22,7 @@ import type {
   ThemeServiceStart,
 } from '@kbn/core/public';
 import type { LensPublicStart } from '@kbn/lens-plugin/public';
+import { type GetMlSharedImportsReturnType } from '@kbn/ml-plugin/public';
```

Comment:
As you mentioned in Slack, it's possible we might have to do some refactoring here, so we are not importing these components statically... I suspect there might be some issue of importing code from ml, but then we are embedding these aiops component back into ml. This is working for transform because the import is only one way.

Reply:
To me this looks fine for now for this PR. Here we just import the type and the actual hook+component get passed down from the `ml` plugin into the `aiops` wrapping component so it's not a circular dependency. Longer term we might want to move everything that's shared like this from the `ml` plugin to packages though.

Reply:
Yes, @walterra is right, importing type is safe and doesn't affect the bundle size. 
But in general, we should not share components, services and hooks statically. Instead, it should happen in the plugin `setup`. I'll create an issue for that.

Reply:
FYI I had to remove this import in https://github.com/elastic/kibana/pull/155177/commits/eb9f239ea957e443e1bd3336ecfd8dbc9b5833e4 because it caused circular dependency, as ml plugin already contains a reference to aiops plugin, we are not allowed to reference it from the aiops plugin.

Reply:
LGMT 🎉